media/js/base/core-datalayer.js

```diff
@@ -25,6 +25,33 @@ if (typeof Mozilla == 'undefined') {
         return pageId ? pageId : pathName.replace(/\/[^\/]*/, '');
     };
 
+    /** Returns whether page has download button.
+    * @param {String} path - URL path name fallback if page ID does not exist.
+    * @return {String} string.
+    */
+    Analytics.pageHasDownload = function() {
+        return $('[data-download-os]').length ? 'true' : 'false';
+    };
+
+    /** Returns whether page has video.
+    * @param {String} path - URL path name fallback if page ID does not exist.
+    * @return {String} string.
+    */
+    Analytics.pageHasVideo = function() {
+        return ($('video').length || $('iframe[src*="youtube"]').length) ? 'true' : 'false';
```

Comment:
Nit: let's make this a little bit more explicit and check for `iframe[src^="https://www.youtube"]`. This should cover both regular and nocookie domains, but also ensures it appears at the start of the `src` value.